debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/AbstractMysqlFieldReader.java

```diff
@@ -40,38 +40,46 @@ public Object readField(ResultSet rs, int columnIndex, Column column, Table tabl
         if (column.jdbcType() == Types.TIME) {
             return readTimeField(rs, columnIndex);
         }
-        else if (column.jdbcType() == Types.DATE) {
+
+        if (column.jdbcType() == Types.DATE) {
```

Comment:
Why do you replace `else if` with `if`, is there any issue with it?

Reply:
I was about to mention this.  It looks like the PR uses a slightly different approach to resolving the value converter near the end of the if blocks.  If we don't want to inline the `.getValueConverter(table.id(), column).isPresent()` call in the if check itself, then my suggestion would be to pull the original last `else if` and `else` into a single `else` block, calculate the value of `hasConverterForColumn` and then simply use a nested if-else block to keep the changes as localized as possible, i.e.:

```
else {
  boolean hasConverterForColumn = connectorConfig.customConverterRegistry()
                .getValueConverter(table.id(), column)
                .isPresent();
        if (hasConverterForColumn && TEXT_DATATYPES.contains(column.typeName())) {
            try {
                String columnData = rs.getString(columnIndex);
                if (columnData != null) {
                    return columnData.getBytes(column.charsetName());
                }
            }
            catch (UnsupportedEncodingException e) {
                logger.warn("Unsupported encoding '{}' for column '{}', sending value as String", e.getMessage(), column.name());
            }
        }
        return rs.getObject(columnIndex);
}
```

Reply:
@jpechane The `else if` is reduced to just an `if` statement as all `else if` and `else` branches return value. There is no value in having a chained `else if` conditional statement if all branches return value.

To contrast what my change is about:
```java
if (a == 1) {
	return one;
} 
else if (a == 2) {
	return two;
} 
else {
	return three;
}
```
vs
```java
if (a == 1) {
	return one;
}
if (a == 2) {
	return two;
}
return three;
```

Both approaches do the same, and the second approach doesn't have an `else` condition. I'd consider using `else if` only in cases where the branches do not return value but do state modification.

I've removed the `else` conditional because it will be the last thing evaluated and returns, so there is no value in nesting.

This code change does not change the flow of execution of statements and, in my opinion, makes for a more readable code as it reduces the nesting and chained conditionals.

I'd love your feedback and I'm crossing my fingers we can get this merged.

Reply:
@nenad ok, having a second thought for it it is not exactly necessary but you are right that with `return` in place this is not exactly `switch` simulation.
It now looks more the method style I prefer - do as much you can do at top and return early. So I am merging it.
Thanks a lot.